pygfx/resources/_texturemap.py

```diff
@@ -0,0 +1,119 @@
+from ..utils.trackable import Trackable
+
+
+class TextureMap(Trackable):
+    """A texture map.
+
+    TextureMap is used to define how a texture should be used in a material.
+    including which uv channel to use, filtering, and wrapping
+
+    Parameters
+    ----------
+    texture : Texture
+        The texture to use for this map.
+    channel : int, optional
```

Comment:
I'm a bit worried about confusion with color channels (e.g. #882). Would it make sense to rename this to something like `coords` or `coord_index`? Or is `channel`  a common name in this context?

Idea: maybe we could do `coords="texcoords7"`, i.e. it represents the full name on the geometry to use. That would actually be more explicit and simpler.

Reply:
**UV channel** is a specialized term in computer graphics, representing different coordinate sets of a model. To avoid ambiguity, we can rename the attribute as `uv_channel` .

For the parameter, I believe using an index is preferable to using the geometry attribute name (like texcoords7). There are two reasons for this: first, using an index follows common conventions and habits, and second, from the perspective of user code simplicity, `TextureMap(uv_channel=7)` is much more convenient than `TextureMap(uv_channel="texcoords7")` :)

Reply:
`uv_channel`  sounds good!